src/libs/actions/BankAccounts.js

```diff
@@ -270,12 +272,362 @@ function fetchUserWallet() {
         });
 }
 
+let previousACHData = {};
+Onyx.connect({
+    key: ONYXKEYS.REIMBURSEMENT_ACCOUNT,
+    callback: (val) => {
+        previousACHData = lodashGet(val, 'achData', {});
+    },
+});
+
+/**
+ * Navigate to a specific step in the VBA flow
+ *
+ * @param {String} stepID
+ * @param {Object} achData
+ */
+function goToWithdrawalAccountSetupStep(stepID, achData) {
+    const newACHData = {...previousACHData};
+
+    // If we go back to Requestor Step, reset any validation and previously answered questions from expectID.
+    if (!newACHData.useOnfido && stepID === CONST.BANK_ACCOUNT.STEP.REQUESTOR) {
+        delete newACHData.questions;
+        delete newACHData.answers;
+        if (lodashHas(achData, CONST.BANK_ACCOUNT.VERIFICATIONS.EXTERNAL_API_RESPONSES)) {
+            delete newACHData.verifications.externalApiResponses.requestorIdentityID;
+            delete newACHData.verifications.externalApiResponses.requestorIdentityKBA;
+        }
+    }
+
+    // When going back to the BankAccountStep from the Company Step
+    if (newACHData.currentStep === CONST.BANK_ACCOUNT.STEP.COMPANY && stepID === CONST.BANK_ACCOUNT.STEP.BANK_ACCOUNT) {
+        newACHData.subStep = 'manual';
+    }
+
+    Onyx.merge(ONYXKEYS.REIMBURSEMENT_ACCOUNT, {achData: {...newACHData, ...achData, currentStep: stepID}});
+}
+
+/**
+ * Fetch the bank account currently being set up by the user for the free plan if it exists.
+ */
+function fetchFreePlanVerifiedBankAccount() {
+    Onyx.merge(ONYXKEYS.REIMBURSEMENT_ACCOUNT, {loading: true});
+    let bankAccountID;
+    let bankAccount;
+    let kycVerificationsMigration;
+    API.Get({
```

Comment:
Instead of doing four API requests one-after-another, we can send them off asynchronously using (our polyfill for) [`promiseAllSettled`](https://github.com/Expensify/Expensify.cash/tree/957815e5539a76de62da4568a6727d7f8c956467/src/libs/promiseAllSettled). Something like this:

```javascript
Onyx.merge(ONYXKEYS.REIMBURSEMENT_ACCOUNT, {loading: true});
promiseAllSettled([
    API.Get({
        returnValueList: 'nameValuePairs',
        name: CONST.NVP.FREE_PLAN_BANK_ACCOUNT_ID,
    }),
    API.Get({
        returnValueList: 'nameValuePairs',
        name: 'expensify_migration_2020_04_28_RunKycVerifications',
    }),
    API.Get({returnValueList: 'bankAccountList'}),
    API.Get({
        returnValueList: 'nameValuePairs',
        name: CONST.NVP.ACH_DATA_THROTTLED,
    }),
])
    .then((responses) => {
        const bankAccountID = lodashGet(responses, `[0].nameValuePairs.${CONST.NVP.FREE_PLAN_BANK_ACCOUNT_ID}`, '');
        const kycVerificationsMigration = lodashGet(
            responses,
            '[1].nameValuePairs.expensify_migration_2020_04_28_RunKycVerifications',
            '',
        );
        const bankAccountJSON = _.find(responses[2].bankAccountList, account => (
            account.bankAccountID === bankAccountID
        ));
        const bankAccount = bankAccountJSON ? new BankAccount(bankAccountJSON) : null;
        const throttledDate = lodashGet(responses[3].nameValuePairs, CONST.NVP.ACH_DATA_THROTTLED, '');

        // Next we'll build the achData and save it to Onyx
        // If the user is already setting up a bank account we will continue the flow for them
        let currentStep = previousACHData.currentStep;
        ...
        ...
    })
```

Reply:
Seems reasonable to me. These requests are pretty fast, but I like containing all the logic in one `.then()`